integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/ServiceBusRestClient.java

```diff
@@ -0,0 +1,181 @@
+package no.difi.meldingsutveksling.nextmove;
+
+import com.google.common.hash.Hashing;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import lombok.extern.slf4j.Slf4j;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.dokumentpakking.xml.Payload;
+import no.difi.meldingsutveksling.domain.MeldingsUtvekslingRuntimeException;
+import no.difi.meldingsutveksling.domain.sbdh.EduDocument;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import org.eclipse.persistence.jaxb.JAXBContextFactory;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.xml.transform.StringSource;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+@Component
+@Slf4j
+public class ServiceBusRestClient {
+
+    private static final String NEXTMOVE_QUEUE_PREFIX = "nextbestqueue";
+
+    private ServiceRegistryLookup sr;
+    private IntegrasjonspunktProperties props;
+    private String localQueuePath;
+    private JAXBContext jaxbContext;
+
+    public ServiceBusRestClient(ServiceRegistryLookup sr,
+                                IntegrasjonspunktProperties props) throws JAXBException {
+        this.sr = sr;
+        this.props = props;
+
+        this.localQueuePath = NEXTMOVE_QUEUE_PREFIX+
+                props.getOrg().getNumber()+
+                props.getNextbest().getServiceBus().getMode();
+        this.jaxbContext = JAXBContextFactory.createContext(new Class[]{EduDocument.class, Payload.class, ConversationResource.class}, null);
+    }
+
+    public void sendMessage(byte[] message, String queuePath) {
+        String resourceUri = format("https://%s.servicebus.windows.net/%s/messages",
+                props.getNextbest().getServiceBus().getNamespace(), queuePath);
+        URI uri = convertToUri(resourceUri);
+
+        String auth = createAuthorizationHeader(resourceUri);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", auth);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Authorization" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)